drivers/gpio/gpio_pca95xx.c

```diff
@@ -470,16 +622,26 @@ static int gpio_pca95xx_init(struct device *dev)
 	struct gpio_pca95xx_drv_data * const drv_data =
 		(struct gpio_pca95xx_drv_data * const)dev->data;
 	struct device *i2c_master;
+	int ret;
 
 	/* Find out the device struct of the I2C master */
 	i2c_master = device_get_binding((char *)config->i2c_master_dev_name);
 	if (!i2c_master) {
 		return -EINVAL;
 	}
 	drv_data->i2c_master = i2c_master;
+	drv_data->exp_dev = dev;
 
 	k_sem_init(&drv_data->lock, 1, 1);
 
+#ifdef CONFIG_PCA95XX_INTERRUPT_SUPPORT
+	ret = gpio_pca95xx_interrupt_init(dev);
```

Comment:
Make this:

    int ret = gpio....

and remove the declaration at the start of the function.  It produces diagnostics when interrupt support is not enabled.